### PR TITLE
Merge test

### LINKED ISSUE #118: Merge thirtyinch-test in thirtyinch
Currently we have a extra `thirtyinch-test` dependency which includes just [a single file](https://github.com/grandcentrix/ThirtyInch/blob/master/test/src/main/java/net/grandcentrix/thirtyinch/test/TiPresenterInstructor.java).
Because it is way more overhead to maintain these extra package we agreed that we want to merge it into the "[main](https://github.com/grandcentrix/ThirtyInch/tree/master/thirtyinch)" `thirtyinch` module.

We want to have a similar idea like RxJava in tests.
Just calling: 
```kotlin
val presenterInstructor = MyPresenterImpl().test()
```
Currently we are not really sure if we really want to return the `PresenterInstructor` or another (helper) object which hold the real presenter and the instructor...
Maybe later we want to add more functionality to it which can nicely delegate the "actions" (or maybe `asserts` 🤔 ) to the correct object...

### PR DESCRIPTION
**Pull request by StefMa**
This will fix #118 

Currently I've just copied the Instructor code into the `TiTestPresenter`, added some tests (why the TestPresener is needed - not testing it themself) and added a `test()` method...

### CODE REVIEW

File: thirtyinch/src/test/java/net/grandcentrix/thirtyinch/test/TiTestPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                sendToView(new ViewAction<MockTiView>() {
+                    @Override
+                    public void call(MockTiView tiView) {
+                        tiView.helloWorld();
+                    }
+                });
+            }
+        };
+        mockTiPresenter.create();
+        mockTiView = new MockTiView() {
```
Comment (passsy) [inline thirtyinch/src/test/java/net/grandcentrix/thirtyinch/test/TiTestPresenterTest.java]: use `Mockito.mock(MockTiView.class);` and verify that `helloWorld()` was called.

**Comment by StefMa** [comment]
For me the changes look good.
I think we can ignore codecov because of -0,02% 🙄 

Anything missing here? Otherwise I would say :shipit:

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java [hunk 2]
```diff
@@ -386,6 +386,10 @@ public void setUiThreadExecutor(@Nullable final Executor uiThreadExecutor) {
         mUiThreadExecutor = uiThreadExecutor;
     }
 
+    public TiTestPresenter<V> test() {
```
Comment (passsy) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java]: Thought about this method a lot, especially if it is extensible enough for the future. It is. When my presenter requires additional setup for testing one can override this method. That's great 🚢

Comment (passsy) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java]: Missing documentation for `test()`. Copy from `TiTestPresenter`

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/test/TiTestPresenter.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        switch (state) {
+            case INITIALIZED:
+                mPresenter.create();
+                break;
+            case VIEW_DETACHED:
+                // already there
+                break;
+            case VIEW_ATTACHED:
+                mPresenter.detachView();
+                mPresenter.setUiThreadExecutor(null);
```
Comment (passsy) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/test/TiTestPresenter.java]: task for Hacktoberfest: Add tests for `TiTestPresenter`

**Pull request merged**